src/test/app/SetTrust_test.cpp

```diff
@@ -17,50 +17,194 @@
 */
 //==============================================================================
 #include <ripple/test/jtx.h>
+#include <ripple/protocol/TxFlags.h>
+#include <ripple/protocol/JsonFields.h>
 
 namespace ripple {
+
 namespace test {
 
 class SetTrust_test : public beast::unit_test::suite
 {
 public:
-    void testFreeTrustlines()
+
+
+    void testFreeTrustlines(bool thirdLineCreatesLE, bool createOnHighAcct)
     {
-        testcase("Allow 2 free trust lines before requiring reserve");
+        testcase(std::string("Allow two free trust lines before requiring reserve; third line ") +
+            (thirdLineCreatesLE ? "creates ledger entry" : "adds to existing ledgerr entry")
+                + " creator is " + (createOnHighAcct ? "high" : "low") + " account ");
 
         using namespace jtx;
         Env env(*this);
 
-        auto const gwA = Account{ "gatewayA" };
-        auto const gwB = Account{ "gatewayB" };
-        auto const gwC = Account{ "gatewayC" };
-        auto const alice = Account{ "alice" };
+        auto const gwA = Account{ "gwA" };
+        auto const gwB = Account{ "gwB" };
+        auto const acctC = Account{ "acctC" };
+        auto const acctD = Account{ "acctD" };
+
+        auto const & creator  = createOnHighAcct ? acctD : acctC;
```

Comment:
Just from curiosity, how did you determine that acctD is higher than acctC?  Not that it really matters, since one is higher than the other and you test both cases...